common/src/main/java/com/microsoft/identity/common/internal/fido/PublicKeyCredentialDescriptor.java

```diff
@@ -20,14 +20,23 @@
 //  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
-package com.microsoft.identity.common.internal.fido
+package com.microsoft.identity.common.internal.fido;
+
+import androidx.annotation.NonNull;
+
+import lombok.Getter;
 
 /**
  * Representation of WebAuthn's PublicKeyCredentialDescriptor.
  * https://w3c.github.io/webauthn/#dictdef-publickeycredentialdescriptor
  */
-@kotlinx.serialization.Serializable
```

Comment:
Do we need to convert all classes into java? would the non-data kotlin class work?

asking because their repo says that they support both. https://github.com/square/moshi

Reply:
Non-data Kotlin classes should still work. The unit tests are written in Kotlin and can run, and I did test calling WebAuthnJsonUtil from some other Kotlin classes (like the PasskeyFidoChallengeHandler, which will come out in a future PR) and it did what it needed to do. The only issue I was thinking about was the challenge object being a kotlin class, but it doesn't throw any errors while referenced in WebAuthnJsonUtil.

Moshi supports both, but for Kotlin, an extra dependency needs to be added: https://github.com/square/moshi#kotlin
Reflection adds a lot in size, and while the codegen option is smaller, from my googling, it slows down build speed.
I was thinking that if I just wrote this part of the feature (WebAuthnJsonUtil and associated data classes) in Java, it could help minimize size increase (only one dependency, no plugin, and it's already in broker) with no known potential impact to build time.

Reply:
i see. surprising that even WebAuthnJsonUtil itself needs to be in java.

got it. thanks for the info.

Reply:
Actually you're right... WebAuthnJsonUtil does not need to be written in Java, as the extra dependencies are specifically for converting Kotlin classes to and from JSON. I think I misunderstood your initial comment. 
Addressed here: 3723fbe9dc03bac93ad9e3a785ec01697d73a1b4